File: app/src/main/java/com/selflearning/chemistree/adapter/PostGameReviewAdapter.java
package com.selflearning.chemistree.adapter;

import android.app.Activity;
import android.content.Context;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.selflearning.chemistree.R;
import com.selflearning.chemistree.models.ReportModels;

import java.util.ArrayList;

public class PostGameReviewAdapter extends RecyclerView.Adapter<PostGameReviewAdapter.PostGameReviewViewHolder> {

    private Context context;
    private Activity activity;

    private ArrayList<ReportModels> arrayList;

    public PostGameReviewAdapter(Context context, Activity activity, ArrayList<ReportModels> arrayList) {
        this.context = context;
        this.activity = activity;
        this.arrayList = arrayList;
    }

    @NonNull
    @Override
    public PostGameReviewViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.rv_post_game_report, parent, false);
        return new PostGameReviewViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull PostGameReviewViewHolder holder, int position) {
        ReportModels reportModels = arrayList.get(position);
        holder.tvName.setText(Html.fromHtml(reportModels.getCorrectName()));
        holder.tvDesc.setText(Html.fromHtml(reportModels.getCorrectDesc()));
    }

    @Override
    public int getItemCount() {
        return (arrayList != null ? arrayList.size() : 0);
    }

    class PostGameReviewViewHolder extends RecyclerView.ViewHolder{

        private TextView tvName;
        private TextView tvDesc;

        public PostGameReviewViewHolder(@NonNull View itemView) {
            super(itemView);
            tvName = itemView.findViewById(R.id.tvReportName);
            tvDesc = itemView.findViewById(R.id.tvReportDesc);
        }
    }
}

File: app/src/main/java/com/selflearning/chemistree/adapter/PostGameReviewAdapter.kt
package com.selflearning.chemistree.adapter

import android.text.Html
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.text.HtmlCompat
import androidx.recyclerview.widget.RecyclerView
import com.selflearning.chemistree.adapter.PostGameReviewAdapter.PostGameReviewViewHolder
//import com.selflearning.chemistree.databinding.RvPostGameReportBinding
import com.selflearning.chemistree.models.ReportModels
import java.util.*

class PostGameReviewAdapter(private val arrayList: ArrayList<ReportModels>) : RecyclerView.Adapter<PostGameReviewViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PostGameReviewViewHolder {
//        val binding = RvPostGameReportBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return PostGameReviewViewHolder(parent)
    }

    override fun onBindViewHolder(holder: PostGameReviewViewHolder, position: Int) {
        val reportModels = arrayList[position]
        holder.bind(reportModels)
    }

    override fun getItemCount(): Int {
        return arrayList.size
    }

    inner class PostGameReviewViewHolder(private val parent: ViewGroup) : RecyclerView.ViewHolder(parent) {

        fun bind(model: ReportModels){
//            binding.tvReportDesc.text = HtmlCompat.fromHtml(model.correctDesc, HtmlCompat.FROM_HTML_MODE_LEGACY)
//            binding.tvReportName.text = Html.fromHtml(model.correctName)
        }
    }
}